local.go

```diff
@@ -59,6 +59,9 @@ func Local(f Function, initX []float64, settings *Settings, method Method) (*Res
 	if method == nil {
 		method = getDefaultMethod(funcInfo)
 	}
+	if method.Needs().Hessian && !method.Needs().Gradient {
```

Comment:
I don't like this. If the method doesn't need the gradient, then it doesn't need the gradient. The method shouldn't have to know it will be used in Local. We can just change line 215 to add an  "|| method.Needs().Hessian".